From 68105da2f498efb549af88a251d938f10ab57fa8 Mon Sep 17 00:00:00 2001
From: LatvianModder <latvianmodder@gmail.com>
Date: Fri, 13 Apr 2018 01:06:18 +0300
Subject: [PATCH 264/296] backup: Fix a system server crash

 * If we get into a state where the backup data is null, system server
   will go down in flames. Abort and kill any active agents if this
   happens.
---
 .../backup/java/com/android/server/backup/BackupManagerService.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/services/backup/java/com/android/server/backup/BackupManagerService.java b/services/backup/java/com/android/server/backup/BackupManagerService.java
index b6fd0729931..278971c7e7b 100644
--- a/services/backup/java/com/android/server/backup/BackupManagerService.java
+++ b/services/backup/java/com/android/server/backup/BackupManagerService.java
@@ -3341,6 +3341,12 @@ public class BackupManagerService implements BackupManagerServiceInterface {
 
                 final String pkgName = mCurrentPackage.packageName;
                 final long filepos = mBackupDataName.length();
+                if (mBackupDataName == null) {
+                    failAgent(mAgentBinder, "Backup data was null: " + mBackupDataName);
+                    addBackupTrace("backup data was null: " + mBackupDataName);
+                    errorCleanup();
+                    return;
+                }
                 FileDescriptor fd = mBackupData.getFileDescriptor();
                 try {
                     // If it's a 3rd party app, see whether they wrote any protected keys
-- 
2.11.0

